Tasks/AppStorePromote/app-store-promote.ts

```diff
@@ -38,19 +68,39 @@ async function run() {
         // Get input variables
         let authType: string = tl.getInput('authType', false);
         let credentials: UserCredentials = new UserCredentials();
+        let apiKey: ApiKey = undefined;
+
+        const createApiKeyFileName = (apiKeyId: string) => {
+            const tempPath = tl.getVariable('Agent.BuildDirectory') || tl.getVariable('Agent.TempDirectory');
```

Comment:
I'd suggest to use `Agent.TempDirectory` or at least swap these variables since `Agent.TempDirectory` is cleaned up after each job and it seems to be safer.

Reply:
That's a good idea, I'll fix it now.

Reply:
@egor-bryzgalov  Fixed in https://github.com/microsoft/app-store-vsts-extension/pull/189/commits/b79c37a1495a08eaf00671b879017554ac815080